frontend/apps/filemanager/filemanagercollection.lua

```diff
@@ -0,0 +1,124 @@
+
+local FileManagerCollection = InputContainer:extend{
+    coll_menu_title = _("Favorites"),
+}
+
+function FileManagerCollection:init()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function FileManagerCollection:addToMainMenu(menu_items)
+    menu_items.collections = {
+        text = self.coll_menu_title,
+        callback = function()
+            self:onShowColl("favorites")
+        end,
+    }
+end
+
+function FileManagerCollection:updateItemTable()
+    -- Try to stay on current page.
+    local select_number = nil
+    if self.coll_menu.page and self.coll_menu.perpage then
+        select_number = (self.coll_menu.page - 1) * self.coll_menu.perpage + 1
+    end
+    self.coll_menu:switchItemTable(self.coll_menu_title,
+        ReadCollection:prepareList(self.coll_menu.collection), select_number)
+end
+
+function FileManagerCollection:onMenuHold(item)
+    local collfile_dialog = nil
+    local buttons = {
+        {
+            {
+                text = _("Sort"),
+                callback = function()
+                    UIManager:close(collfile_dialog)
+                    local item_table = {}
+                    for i=1, #self._manager.coll_menu.item_table do
+                        table.insert(item_table, {text = self._manager.coll_menu.item_table[i].text, label = self._manager.coll_menu.item_table[i].file})
+                    end
+                    local SortWidget = require("ui/widget/sortwidget")
+                    local sort_item
+                    sort_item = SortWidget:new{
+                        title = _("Sort favorites book"),
+                        item_table = item_table,
+                        callback = function()
+                            local new_order_table = {}
+                            for i=1, #sort_item.item_table do
+                                table.insert(new_order_table, {
+                                    file = sort_item.item_table[i].label,
+                                    order = i
+                                })
+                            end
+                            ReadCollection:writeCollection(new_order_table, self._manager.coll_menu.collection)
+                            self._manager:updateItemTable()
+                        end
+                    }
+                    UIManager:show(sort_item)
+
+                end,
+            },
+            {
+                text = _("Remove from collection"),
+                callback = function()
+                    ReadCollection:removeItem(item.file, self._manager.coll_menu.collection)
+                    self._manager:updateItemTable()
+                    UIManager:close(collfile_dialog)
+                end,
+            },
+        },
+        {
+            {
+                text = _("Book information"),
+                enabled = FileManagerBookInfo:isSupported(item.file),
+                callback = function()
+                    FileManagerBookInfo:show(item.file)
```

Comment:
Later, you might want to tweak the onHold dialog when the user uses a CoverBrowser mode, like we did for History in:
https://github.com/koreader/koreader/blob/a5419439ce8115a8571a8c78c6735618c82c270b/plugins/coverbrowser.koplugin/covermenu.lua#L398-L400
so we have the other buttons like View book description, View full size cover...